### PR TITLE
Add remove

### PR DESCRIPTION
**Pull request by Mucyosoda**
This project will lay foundations for the Bookstore website.
React and Redux app was built and set up routing using React Router.
This app is able to add and delete books from the booklist by clicking on the delete button
the project built-in :

React
Redux
webpack
babelrc
linter

### CODE REVIEW

File: src/components/books.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        <label htmlFor="author">
+          Author:
+          <input type="text" name="author" required />
+        </label>
+        <label htmlFor="year">
+          Year:
+          <input type="text" name="year" required />
+        </label>
+        <input type="submit" value="Add Book" />
+      </form>
```
Comment (TedLivist) [inline src/components/books.js]: - [ ] In keeping with the SOLID principle, let's refactor this **`books`** component to be a lot neater. There should be a separate component responsible for adding a book (eg BookForm/AddBook/CreateBook or any name). Kindly extract this form into a new component and import the component into **`books`** instead.

- [ ] Your inputs should be controlled and states should be the only source of truth. You can refer to the first paragraph of this [article](https://reactjs.org/docs/forms.html) to get an idea of what this means.

  As listed in the project requirements (see image below) your component for adding a new book should have a local `state` for `title` and `author`. And that local state should be updated whenever a user updates the title and author inputs with `onChange`. This state is what should be dispatched. You can refer to this [article](https://dmitripavlutin.com/controlled-inputs-using-react-hooks/) for a quick understanding on how to create controlled components with React hooks.

  muth-req

File: src/components/books.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      <ul>
+        {bookArr.map(((item) => (
+          <div key={item.id} className="book-item">
+            <li>{`${item.title}, ${item.author}, ${item.year}`}</li>
+            <button type="button" className={item.id} onClick={(e) => dispatch(removeBook({ id: e.target.className }))}>delete</button>
```
Comment (TedLivist) [inline src/components/books.js]: - [ ] Again, let's clean up the **`books`** component. As stated in the requirement, there should be a component responsible for removing a book - which is the same component responsible for a single book (eg Book/OneBook/SingleBook or whatever name). Kindly extract these two lines into a new component responsible for a single book and pass in the book data as a prop instead. Then render the component here in the map iteration.

  muyo-book

File: src/Redux/books/books.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+export const removeBook = (payload) => ({
+  type: 'REMOVE_BOOK',
+  payload,
+});
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case 'ADD_BOOK':
+      return [...state, action.payload];
+    case 'REMOVE_BOOK':
```
Comment (TedLivist) [inline src/Redux/books/books.js]: - [ ] As stated in the requirements, your actions should be defined using the **ducks pattern**. Kindly defined your `Action Types` to be constants (see image) and then use those constants in your `Action Creators` and `reducer`.

  muyo-ducks

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Mucyosoda,

Good job so far!
✔️ Good documentation
✔️ Redux is well set up
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by omar-labana** [comment]
Hi @Mucyosoda, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**